x-pack/plugins/dashboard_enhanced/public/services/drilldowns/dashboard_to_dashboard_drilldown/drilldown.tsx

```diff
@@ -153,7 +150,7 @@ export class DashboardToDashboardDrilldown
       }
     }
 
-    return getUrlGenerator(DASHBOARD_APP_URL_GENERATOR).createUrl({
+    return this.params.getDashboardUrlGenerator().createUrl!({
```

Comment:
huh, having a hard time following this without pulling down the code... why would this possibly be undefined?

Reply:
I just realised a difference... 
When I started exporting directly and using directly it is `UrlGeneratorsDefinition`
But when we access from `share` plugin registry it is `UrlGeneratorContract`. 
So url generator registry wrap the definition and adds additional logic. Which means this "refactor" is not 1 to 1 functionality match. 

Not sure how to process from here in this case,

Reply:
ahhhh.  Yea this is the problem discussed in that issue https://github.com/elastic/kibana/issues/62815.  However, this PR has a solution for it: https://github.com/elastic/kibana/pull/62998 that may work.  You'd have to change it to `registerUrlGenerator` returns the getter that will return the contract in the start life cycle.

Reply:
Thanks!

`share` plugin is optional in dashboard. So I guess this also means that dashboardUrlGenerator will be optional on `dashboard` plugin contract?

Reply:
somewhat of a bummer, but that share plugin contains the code for handling migrations, automatically which is why the `Definition` becomes a `Contract`.  This may change when Persistable state registries becomes a formal, generic service.  I would take the path of least resistance for now (require share plugin for this functionality), and we can revisit when we know more about how the persistable state registries will work.

Reply:
@stacey-gammon please take a look 🙏 
hope I followed the proposed solution correctly.

This needs further clean up: like removing global mappings and fixing how ML uses it, but I don't want to do this in this pr, because it points to drilldowns branch